In [1]:
import numpy as np
import scipy.stats as stats

# Домашняя работа по уроку 7. Непараметрические тесты

---------

## Задача 1

Даны две независимые выборки. Не соблюдается условие нормальности

x1 380,420, 290

y1 140,360,200,900

Сделайте вывод по результатам, полученным с помощью функции, 

имеются ли статистические различия между группами?

### Решение

Сравниваем 2 независимые выборки с помощью критерия Манна-Уитни

In [2]:
x1 = np.array([380, 420, 290])
y1 = np.array([140, 360, 200, 900])

stats.mannwhitneyu(x1, y1)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

Ответ: pvalue = 0.629 > α = 0.05, =>, следовательно верна 

нулевая гипотеза об отсутствии статистических различий между группами

------------

## Задача 2

Исследовалось влияние препарата на уровень давления пациентов. 

Сначала измерялось давление до приема препарата, потом через 10 минут и через 30 минут. 

Есть ли статистически значимые различия между измерениями давления? 

В выборках не соблюдается условие нормальности.

1е измерение до приема препарата: 150, 160, 165, 145, 155

2е измерение через 10 минут: 140, 155, 150, 130, 135

3е измерение через 30 минут: 130, 130, 120, 130, 125

### Решение

Сравниваем 3 выборки, анализируя повторные измерения,

с помощью критерия Фридмана

In [3]:
arr1 = np.array([150, 160, 165, 145, 155])
arr2 = np.array([140, 155, 150, 130, 135])
arr3 = np.array([130, 130, 120, 130, 125])

stats.friedmanchisquare(arr1, arr2, arr3)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

Ответ: pvalue = 0.008 < α = 0.05, =>, следовательно, нулевую гипотезу об отсутствии различий отвергаем,

т.е. различия между измерениями есть, препарат влияет на уровень давления пациентов.

----------

## Задача 3

Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было. 

Есть ли статистически значимые различия между измерениями давления?

### Решение

Сравниваем 2 зависимые выборки, т.к. имеем дело с одними и теми же пациентами,

с помощью критерия Уилкоксона

In [4]:
arr1 = np.array([150, 160, 165, 145, 155])
arr2 = np.array([140, 155, 150, 130, 135])

stats.wilcoxon(arr1, arr2)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

Ответ: pvalue = 0.063 > α = 0.05, =>, следовательно, нулевая гипотеза об отсутствии различий верна.

--------------

## Задача 4

Даны 3 группы учеников плавания. Не соблюдается условие нормальности.

В 1 группе время на дистанцию 50 м составляют:

56, 60, 62, 55, 71, 67, 59, 58, 64, 67

Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53

Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54

Есть ли статистически значимые различия между группами?

### Решение

Сравниваем 3 независимые выборки, т.к. имеем дело с разными участниками,

с помощью критерия Крускала-Уоллиса

In [5]:
gr1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
gr2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
gr3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

stats.kruskal(gr1, gr2, gr3)

KruskalResult(statistic=5.465564058257224, pvalue=0.06503809985904942)

Ответ: pvalue = 0.065 > α = 0.05, =>, следовательно, нулевая гипотеза об отсутствии различий между группами верна.

-----------

## Задача 5

Заявляется, что партия изготавливается со средним арифметическим 2,5 см. 

Проверить данную гипотезу, если известно, 

что размеры изделий подчинены нормальному закону распределения. 

Объем выборки 10, уровень статистической значимости 5%

2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

### Решение

1) Формулируем гипотезы:

H0 : mu = 2.5 - предполагаем, что размер изделий составляет 2.5 см.

H1 : mu != 2.5 - предполагаем, что размер изделий не равен 2.5 см.

α = 0,05 - статистический уровень значимости обозначен условиями задачи.

Нам не известна дисперсия, значит, будем использовать для проверки гипотезы t-критерий:

Найдем наблюдаемое значение t-критерия:

In [6]:
mu = 2.5
n = 10
arr = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])
alpha = 0.05

# Среднее значение по выборке
mean = np.mean(arr)
mean

2.5279999999999996

In [7]:
# Среднее квадратичное отклонение по выборке
std = np.std(arr, ddof=1)
std

0.1572542173961923

In [8]:
# Наблюдаемое значение t-критерия
t = (mean - mu) / (std / np.sqrt(n))
t

0.5630613661802959

Найдем табличное значение критерия Стьюдента для выборки из 10 изделий и α = 0,05

In [9]:
t1 = stats.t.ppf(alpha/2, n - 1)
t2 = stats.t.ppf(1 - alpha/2, n - 1)
t1, t2

(-2.262157162740992, 2.2621571627409915)

In [ ]:
# \\\\\\\\\\                          /////////////
# ----------|----------|-------------|-------------
#          tкр        tн            tкр
#        -2.26       0.56           2.26

Таким образом, видим, что наблюдаемое значение t-критерия расположено

за пределами области критического значения t-критерия.

Поэтому, мы считаем верной гипотезу H0 - средний размер изделий составляет 2.5 см.